src/buildblock/recon_array_functions.cxx

```diff
@@ -224,48 +225,50 @@ void divide_and_truncate(Viewgram<float>& numerator,
       };
 
 #else
-      if(b<bs+rim_truncation_sino ||
-	 b>be-rim_truncation_sino ) 
-	{
-	  numerator[r][b] = 0;
-	}
+      if(b<bs+rim_truncation_sino || b>be-rim_truncation_sino )
+	    numerator[r][b] = 0;
       else
-	{
-	  float& num = numerator[r][b];
-	  if (num<=small_value) // KT Feb2011 was "num<small_value", resulting in a BUG if the whole numerator viewgram was zero
-	    { 
-	      // we think num was really 0 
-	      // (we compare with small_value due to rounding errors)
-	      // this case includes 0/0, but also num<0	     
-	      num = 0;
-	      if (num<0) count2++;
-	    }
-	  else
-	    {
-	      const float max_quotient = 10000.F;
-	      const float denom = denominator[r][b];
-	      // set quotient to min(numerator/denominator, max_quotient)
-	      // a bit tricky to avoid division by 0	  
-	      // we do this by effectively using
-	      // new_denom = max(denominator[r][b], max_quotient/num)
-	      // Note that this includes the case if a negative denominator
-	      // (in case somebody forward projects an image with negatives)
-	      if (num > max_quotient*denom)
-		{
-		  // cancel singularity
-		  count++;
-		  if (log_likelihood_ptr != NULL) 
-		    sub_result -= double(num*log(num/max_quotient));
-		  num = max_quotient;
-		}
-	      else
-		{
-		  if (log_likelihood_ptr != NULL) 
-		    sub_result -= double(num*log(denom));
-		  num = num/denom;
-		}
-	    }
-	}
+      {
+        if (numerator[r][b] <= small_value) // KT Feb2011 was "num<small_value", resulting in a BUG if the whole numerator viewgram was zero
+        {
+          // we think num was really 0
+          // (we compare with small_value due to rounding errors)
+          // this case includes 0/0, but also num<0
+          numerator[r][b] = 0;
+          if (numerator[r][b]<0)
+            count2++;
+        }
+        else
+        {
+          const float max_quotient = 10000.F;
+          // set quotient to min(numerator/denominator, max_quotient)
+          // a bit tricky to avoid division by 0
+          // we do this by effectively using
+          // new_denom = max(denominator[r][b], max_quotient/num)
+          // Note that this includes the case if a negative denominator
+          // (in case somebody forward projects an image with negatives)
+          if (numerator[r][b] > max_quotient*denominator[r][b])
+          {
+            // cancel singularity
+            count++;
+            if (log_likelihood_ptr != NULL)
+            {
+              // Substitute denominator[r][b] (i.e. ybar) for num / max_quotient (i.e. y/ max_quotient)
+              sub_result += compute_data_fit_term(numerator[r][b], numerator[r][b] / max_quotient, use_KL_divergence, small_value);
```

Comment:
This `compute_data_fit_term` call is different to how `accumulate_loglikelihood` handles it.

Reply:
I haven't checked, but obviously we should have only one way/function to do things (or at least make sure the result is the same)

Reply:
In that case this should be 
```
sub_result += compute_data_fit_term(numerator[r][b], denominator[r][b], use_KL_divergence, small_value);
```
regardless of `if (numerator[r][b] > max_quotient*denominator[r][b])`. Log-likelihood and KL divergence should not have special cases for divide by zeros?